### PR TITLE
Upgrade to Docusaurus 2

### LINKED ISSUE #2261: Migrate Website to Docusaurus 2 Alpha
Hi there! The Docusaurus team is currently developing Docusaurus 2 and are almost done with an MVP that doesn't include translations features, which means there's feature parity with your website's current usage of Docusaurus.

Docusaurus 2 brings about many improvements:

- Client-side rendering with prerendering (site renders without JavaScript!)
- More flexible appearance - Customizable layouts, CSS modules, etc
- Embeddable interactive React components within markdown via MDX

I will be helping you migrate and be the POC throughout the migration process. Let me know what you think!

cc Draft JS team - @claudiopro @mrkev 
cc Docusaurus side - @endiliey @wgao19 @lex111

### PR DESCRIPTION
**Pull request by yangshun**
**Summary**

Fixes #2261. This should be merged after #2267 (how I wish there were stacked diffs on GitHub)

Migrate the website to Docusaurus 2 which provides the following benefits specific to Draft.js:

- Consistency with the other FB Docusaurus websites
- Easy way to import Draft.js (bundle via webpack, no more ugly client-side script tags)
- Nicer CSS framework
- Dark mode (disabled for now until I sort out the dark mode styling for the editor)

Changes made:

- I also changed the theme color of Draft.js to be lighter brown so that the contrast is better in general (much better on dark mode). Let me know if this is not ok.

**Test Plan**

Screen Shot 2019-11-30 at 9 16 32 PM
Screen Shot 2019-11-30 at 9 16 39 PM

Try it out here - https://build-swart-eight.now.sh

### CODE REVIEW

File: website/src/components/DraftEditorExample/css/example.css [hunk 1]
```diff
@@ -0,0 +1,100 @@
+/**
```
Comment (yangshun) [inline website/src/components/DraftEditorExample/css/example.css]: This file already exists in the D1 version, just renamed.

File: website/src/components/DraftEditorExample/index.js [hunk 2]
```diff
@@ -180,105 +178,41 @@ const BlockStyleControls = (props) => {
...
           onToggle={props.onToggle}
           style={type.style}
         />
-      )}
+      ))}
     </div>
   );
 };
 
-ReactDOM.render(
```
Comment (yangshun) [inline website/src/components/DraftEditorExample/index.js]: This is transferred into a separate component.

**Comment by claudiopro** [comment]
This is looking good - I'd prefer to leave the theme color the same as it affects branding (e.g. Twitter account, sticker designs etc.) and would require broader coordination. The project is indeed screaming for better branding and would love to plan something in H1. Let's keep the impact of dark mode on theme colors in a separate conversation for now.

**Comment by claudiopro** [comment]
I think it's okay to leave the current theme color for dark mode for the time being, but I like that you're trying to refresh the look of the website 😃 let's chat about the dark theme for the editor itself on a separate issue.

Screen Shot 2019-12-03 at 11 33 54 AM
Screen Shot 2019-12-03 at 11 33 49 AM

File: website/src/css/custom.css [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ */
+
+ :root {
+  --ifm-color-primary: #bf5934;
+  --ifm-color-primary-dark: rgb(172, 80, 47);
+  --ifm-color-primary-darker: rgb(162, 76, 44);
+  --ifm-color-primary-darkest: rgb(134, 62, 36);
+  --ifm-color-primary-light: rgb(201, 114, 82);
+  --ifm-color-primary-lighter: rgb(210, 139, 113);
+  --ifm-color-primary-lightest: rgb(223, 172, 154);
```
Comment (claudiopro) [inline website/src/css/custom.css]: Please revert these theme colors to the previous value `rgb(132, 49, 49)` and derived variants.

File: website/static/img/draftjs-logo.svg [hunk 4]
```diff
@@ -1,4 +1,4 @@
...
-</svg>
\ No newline at end of file
+  <path d="M91.41 40.632v-.13c0-.176-.044-.307-.044-.482l-.133-.526c0-.044-.045-.044-.045-.087-.045-.175-.134-.307-.223-.438v-.044a3.837 3.837 0 0 0-.267-.438l-.09-.087a2.849 2.849 0 0 0-.356-.394L71.541 19.621 52.829 1.236c-.134-.131-.267-.219-.401-.35L52.338.8c-.133-.088-.311-.176-.445-.263h-.044c-.134-.088-.312-.131-.446-.175-.044 0-.044-.044-.089-.044L50.78.186c-.179-.044-.312-.044-.49-.044H12.599c-2.228 0-4.01 1.75-4.01 3.94v91.924c0 2.188 1.782 3.94 4.01 3.94H87.446c2.227 0 4.01-1.752 4.01-3.94V40.764c-.045-.044-.045-.088-.045-.132zm-37.423-9.367V13.537l9.044 8.886-9.044 8.842zm14.702-3.327l9.044 8.886H59.69l9-8.886zM16.564 7.934h29.404v28.89H16.564V7.934zm0 36.77h29.404v47.275H16.564V44.703zM83.39 91.978H53.987V44.703h29.404V91.98z" fill="#bf5934" fill-rule="nonzero"/>
```
Comment (claudiopro) [inline website/static/img/draftjs-logo.svg]: Please use `rgb(132, 49, 49)` for the color of the logo here.

File: .travis.yml [hunk 5]
```diff
@@ -49,7 +49,7 @@ script:
       git config --global user.name "Travis CI"
       git config --global user.email "travis@reactjs.org"
       echo "machine github.com login reactjs-bot password $GITHUB_TOKEN" > ~/.netrc
-      cd website && yarn install && GIT_USER=reactjs-bot npm run publish-gh-pages
+      cd website && yarn install && GIT_USER=reactjs-bot USE_SSH=true yarn deploy
```
Comment (yangshun) [inline .travis.yml]: This is the new deploy command in Docusaurus v2

File: docs/APIReference-ContentBlock.md [hunk 6]
```diff
@@ -74,7 +74,7 @@ supplied text.
   </li>
   <li>
     <a href="#getcharacterlist">
-      <pre>getCharacterList(): List<CharacterMetadata></pre>
+      <pre>getCharacterList(): List&lt;CharacterMetadata&gt;</pre>
```
Comment (yangshun) [inline docs/APIReference-ContentBlock.md]: Unfortunately we use MDX (JSX in Markdown) now so it thinks `<CharacterMetadata>` is a React component and I have to change it to use the character entity references.

**Comment by yangshun** [comment]
The dark mode for the editor needs more work - the editor text is still the same shade of gray and can hardly be seen, hence I disabled it. Unless you think it's not an issue at all then I shall enable it :D

**Pull request closed without merging**